### PR TITLE
Add TVD Loss Kernel

### PR DESCRIPTION
**Pull request by saurabhkoshatwar**
## Summary
Resolves [#281](https://github.com/linkedin/Liger-Kernel/issues/281). Implements the TVD (Total Variation Distance) kernel by computing both the loss and gradient in the forward pass.

## Testing Done
Implemented tests to verify that the results of the forward and backward passes match the Torch implementation. Additionally, added a script to benchmark the memory usage and speed of the Liger implementation compared to Torch, with the results shown below.

tvd_speed

tvd_memory

- Hardware Type: Nvidia H100 (80GB PCIe)
- [x] run `make test` to ensure correctness
- [x] run `make checkstyle` to ensure code style
- [x] run `make test-convergence` to ensure convergence

### CODE REVIEW

**Comment by saurabhkoshatwar** [comment]
@ByronHsu @qingquansong Please let me know if any changes are required.

File: test/transformers/test_tvd.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    ],
+)
+
+_DTYPE_PARAMS = (
+    "dtype, atol, rtol",
+    [
+        pytest.param(
+            torch.bfloat16,
+            1e-8,
+            5e-2,
```
Comment (yundai424) [inline test/transformers/test_tvd.py:56]: could you help to experiment what is the lowest `rtol` that will not fail this test for bf16? Thanks!

File: test/transformers/test_tvd.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
+from test.utils import supports_bfloat16
+
+import pytest
+import torch
+
+from liger_kernel.transformers.tvd import LigerTVDLoss
+
+
+class TorchTVDLoss(torch.nn.Module):
```
Comment (yundai424) [inline test/transformers/test_tvd.py:9]: I feel it'll be very helpful if we can add ignore index along with this PR to make TVD complete, similar to how JSD is doing it -- https://github.com/linkedin/Liger-Kernel/blob/main/src/liger_kernel/ops/jsd.py

**Review by yundai424 — COMMENTED** [review]
Thanks for the contribution! 😄

**Review by qingquansong — COMMENTED** [review]
Thanks for the efforts! Could you also add this to the init function in transformers folder as well same as JSD? https://github.com/linkedin/Liger-Kernel/blob/main/src/liger_kernel/transformers/__init__.py#L10

File: test/transformers/test_tvd.py (hunk 2 above)
Comment (qingquansong) [reply test/transformers/test_tvd.py:9]: +1 which would be very helpful to cover broader use cases

File: src/liger_kernel/ops/tvd.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        offsets = i + base_offsets
+        mask = offsets < n_cols
+
+        p = tl.load(p_ptr + offsets, mask=mask, other=0.0)
+        q = tl.load(q_ptr + offsets, mask=mask, other=0.0)
+
+        # TVD(P || Q) = 0.5 * |P - Q|
+        tv_loss = 0.5 * tl.abs(p - q)
+
+        grad_res = tl.where(p > q, 0.5, -0.5)
```
Comment (Tcc0403) [inline src/liger_kernel/ops/tvd.py:71]: since we're doing gradients calculation in forward pass already, we can divide gradients by BT (BT * V) based on the reduction mode here to avoid extra calculations in backward pass and saving reduction mode in ctx

File: test/transformers/test_tvd.py (hunk 1 above)
Comment (saurabhkoshatwar) [reply test/transformers/test_tvd.py:56]: @yundai424  approx 5.2e-18 is the lowest rtol that does not fail.
`Test passed with rtol=5.0e-02
Testing with rtol=4.5e-02...
Test passed with rtol=4.5e-02
Testing with rtol=4.0e-02...
Test passed with rtol=4.0e-02
Testing with rtol=3.5e-02...
Test passed with rtol=3.5e-02
Testing with rtol=3.0e-02...
Test passed with rtol=3.0e-02
Testing with rtol=2.5e-02...
Test passed with rtol=2.5e-02
Testing with rtol=2.0e-02...
Test passed with rtol=2.0e-02
Testing with rtol=1.5e-02...
Test passed with rtol=1.5e-02
Testing with rtol=1.0e-02...
Test passed with rtol=1.0e-02
Testing with rtol=5.0e-03...
Test passed with rtol=5.0e-03
Testing with rtol=5.2e-18...
Test passed with rtol=5.2e-18
Testing with rtol=-5.0e-03...
FAILED`

File: src/liger_kernel/ops/tvd.py (hunk 3 above)
Comment (saurabhkoshatwar) [reply src/liger_kernel/ops/tvd.py:71]: done!

File: test/transformers/test_tvd.py (hunk 2 above)
Comment (saurabhkoshatwar) [reply test/transformers/test_tvd.py:9]: done!

File: test/transformers/test_tvd.py (hunk 1 above)
Comment (saurabhkoshatwar) [reply test/transformers/test_tvd.py]: @yundai424 Please ignore previous comment. 1e-6 is the lowest rtol which is not failing.

**Comment by saurabhkoshatwar** [comment]
@yundai424 @Tcc0403 @qingquansong @ByronHsu I have made the changes as per the comments. Let me know if any further changes are required. Thanks!

**Comment by saurabhkoshatwar** [comment]
cc @ByronHsu @qingquansong @lancerts @Tcc0403

**Review by yundai424 — APPROVED** [review]
lgtm
